File: daemon/Models/Database/Journey/TransportType.cs
using NpgsqlTypes;

namespace daemon.Models.Database;

public enum TransportType
{
    [PgName("UNKNOWN")]
    UNKNOWN,
    [PgName("HIGH_SPEED_TRAIN")]
    HIGH_SPEED_TRAIN,
    [PgName("INTERCITY_TRAIN")]
    INTERCITY_TRAIN,
    [PgName("INTER_REGIONAL_TRAIN")]
    INTER_REGIONAL_TRAIN,
    [PgName("REGIONAL_TRAIN")]
    REGIONAL_TRAIN,
    [PgName("CITY_TRAIN")]
    CITY_TRAIN,
    [PgName("SUBWAY")]
    SUBWAY,
    [PgName("TRAM")]
    TRAM,
    [PgName("BUS")]
    BUS,
    [PgName("FERRY")]
    FERRY,
    [PgName("FLIGHT")]
    FLIGHT,
    [PgName("CAR")]
    CAR,
    [PgName("TAXI")]
    TAXI,
    [PgName("SHUTTLE")]
    SHUTTLE,
    [PgName("BIKE")]
    BIKE,
    [PgName("SCOOTER")]
    SCOOTER,
    [PgName("WALK")]
    WALK
}

File: frontend/src/lib/models/TransportType.ts
/**
 * Type of transport.
 *
 * - HIGH_SPEED_TRAIN — High speed train (Hochgeschwindigkeitszug) like ICE or TGV, etc.
 * - INTERCITY_TRAIN — Intercity train (Intercityzug)
 * - INTER_REGIONAL_TRAIN — Interregional train (Interregiozug)
 * - REGIONAL_TRAIN — Regional train (Regionalzug)
 * - CITY_TRAIN — City train (S-Bahn)
 * - SUBWAY — Subway (U-Bahn)
 * - TRAM — Tram (Straßenbahn)
 * - BUS — Bus
 * - FERRY — Ferry (Fähre)
 * - FLIGHT — Flight (Flugzeug)
 * - CAR — Car (Auto)
 * - TAXI — Taxi
 * - SHUTTLE — Shuttle (Ruftaxi)
 * - BIKE — (E-)Bike (Fahrrad)
 * - SCOOTER — (E-)Scooter (Roller)
 * - WALK — Walk (Laufen)
 * - UNKNOWN — Unknown
 * @export
 */
export const TransportType = {
	HighSpeedTrain: "HIGH_SPEED_TRAIN",
	IntercityTrain: "INTERCITY_TRAIN",
	InterRegionalTrain: "INTER_REGIONAL_TRAIN",
	RegionalTrain: "REGIONAL_TRAIN",
	CityTrain: "CITY_TRAIN",
	Subway: "SUBWAY",
	Tram: "TRAM",
	Bus: "BUS",
	Ferry: "FERRY",
	Flight: "FLIGHT",
	Car: "CAR",
	Taxi: "TAXI",
	Shuttle: "SHUTTLE",
	Bike: "BIKE",
	Scooter: "SCOOTER",
	Walk: "WALK",
	Unknown: "UNKNOWN"
} as const;
export type TransportType = (typeof TransportType)[keyof typeof TransportType];
